test/byContract/IexecPoco/IexecPoco1.test.ts

```diff
@@ -279,21 +325,71 @@ describe('IexecPoco1', () => {
             const deal = await iexecPoco.viewDeal(dealId);
             expect(deal.dataset.pointer).to.equal(AddressZero);
             expect(deal.dataset.owner).to.equal(AddressZero);
-            expect(deal.dataset.price.toNumber()).to.equal(0);
+            expect(deal.dataset.price).to.equal(0);
             // BoT size should not be impacted even if the dataset order is the order with the lowest volume
-            expect(deal.botSize.toNumber()).to.equal(volume);
+            expect(deal.botSize).to.equal(botVolume);
         });
 
-        it('[TODO][TEE] Should match orders with restrictions', async () => {});
+        it(`Should match orders with full restrictions in all orders`, async function () {
+            orders.app.datasetrestrict = orders.dataset.dataset;
```

Comment:
Maybe add  `requestorder.workerpool` to emphaze `// requestorder.workerpool is a restriction`